Wbooks/src/app/proptypes/proptypes.js

```diff
@@ -0,0 +1,33 @@
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  optionalArray: PropTypes.array,
+  optionalBool: PropTypes.bool,
+  optionalFunc: PropTypes.func,
+  optionalNumber: PropTypes.number,
+  optionalObject: PropTypes.object,
+  optionalString: PropTypes.string,
+  optionalSymbol: PropTypes.symbol,
+  optionalBook: PropTypes.shape({
+    author: PropTypes.string,
+    genre: PropTypes.string,
+    id: PropTypes.number,
+    image_url: PropTypes.string,
+    publisher: PropTypes.string,
+    title: PropTypes.string,
+    year: PropTypes.string
+  }),
+  optionalArrayOfBooks: PropTypes.arrayOf(
+    PropTypes.shape({
+      author: PropTypes.string,
+      genre: PropTypes.string,
+      id: PropTypes.number,
+      image_url: PropTypes.string,
+      publisher: PropTypes.string,
+      title: PropTypes.string,
+      year: PropTypes.string
+    })
+  )
```

Comment:
```
import { bool, number, string } from 'prop-types';

export const categoryModel = {
  id: number.isRequired,
  name: string.isRequired
};

export const articleModel = {
  id: number.isRequired,
  title: string,
  body: string,
  collapsed: bool
};

export const resultModel = {
  id: number.isRequired,
  sectionId: number.isRequired,
  title: string,
  sectionName: string
};
```
Implement this pattern to this proprTypes files

Reply:
As you can see we just import the necessary propTypes and declare each model by separate so we can reuse them even inside the same file.